hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/upgrade/ScmHAUnfinalizedStateValidationAction.java

```diff
@@ -37,13 +45,22 @@
   public void execute(StorageContainerManager scm) throws Exception {
     boolean isHAEnabled =
         scm.getConfiguration().getBoolean(ScmConfigKeys.OZONE_SCM_HA_ENABLE_KEY,
-        ScmConfigKeys.OZONE_SCM_HA_ENABLE_DEFAULT);
+            ScmConfigKeys.OZONE_SCM_HA_ENABLE_DEFAULT);
 
     if (isHAEnabled) {
-      throw new UpgradeException(String.format("Configuration %s cannot be " +
-          "used until SCM upgrade has been finalized",
-          ScmConfigKeys.OZONE_SCM_HA_ENABLE_KEY),
-          UpgradeException.ResultCodes.PREFINALIZE_ACTION_VALIDATION_FAILED);
+      long lastIndex = scm.getScmHAManager()
```

Comment:
I believe we don't need this, we can check SCM StorageConfig SCM_HA Property.
For Ha enabled cluster, this property is set.

Reply:
Oh I did not realize we were already saving a marker that SCM HA was in use to the disk. That is much better I will use that.